src/community.c

```diff
@@ -3069,26 +3075,33 @@ static int igraph_i_community_multilevel_step(
 
             /* debug("Remove %ld all: %lf Inside: %lf\n", i, -weight_all, -2*weight_inside + weight_loop); */
 
-            /* Find new community to join with the best modification gain */
-            max_q_gain = 0;
-            max_weight = weight_inside;
+            /* Find new community to join */
             n = igraph_vector_size(&links_community);
 
-            for (j = 0; j < n; j++) {
-                long int c = (long int) VECTOR(links_community)[j];
-                igraph_real_t w = VECTOR(links_weight)[j];
-
-                igraph_real_t q_gain =
-                    igraph_i_multilevel_community_modularity_gain(&communities,
-                            (igraph_integer_t) c,
-                            (igraph_integer_t) i,
-                            weight_all, w);
-                /* debug("Link %ld -> %ld weight: %lf gain: %lf\n", i, c, (double) w, (double) q_gain); */
-                if (q_gain > max_q_gain) {
-                    new_id = c;
-                    max_q_gain = q_gain;
-                    max_weight = w;
-                }
+            if (RNG_UNIF01() < probability_q) {
```

Comment:
Instead of finding the best neighbor community in terms of modularity gain, we pick a random neighbor regardless from the gain of modularity.

Reply:
however, a single pass over the whole network should increase the modularity otherwise we are not continuing further iterations.